include/irq.h

```diff
@@ -83,30 +101,21 @@ int32_t irq_ctrl_init(struct irq_ctrl_desc **desc,
 /* Free the resources allocated by irq_ctrl_init(). */
 int32_t irq_ctrl_remove(struct irq_ctrl_desc *desc);
 
-/**
- * @brief Registers a IRQ handling function to irq controller.
- * @param desc - The IRQ controller descriptor.
- * @param irq_id - Interrupt identifier.
- * @param irq_handler - The IRQ handler.
- * @param dev_instance - device instance.
- * @return SUCCESS in case of success, FAILURE otherwise.
- */
-int32_t irq_register(struct irq_ctrl_desc *desc, uint32_t irq_id,
-		     void (*irq_handler)(void *data), void *dev_instance);
+/* Register a callback to handle the irq events */
+int32_t irq_register_callback(struct irq_ctrl_desc *desc, uint32_t irq_id,
+			      struct callback_desc *callback_desc);
+/* Enable specific interrupt */
+int32_t irq_enable(struct irq_ctrl_desc *desc, uint32_t irq_id);
 
 /* Unregisters a generic IRQ handling function */
 int32_t irq_unregister(struct irq_ctrl_desc *desc, uint32_t irq_id);
+/* Disable specific interrupt */
+int32_t irq_disable(struct irq_ctrl_desc *desc, uint32_t irq_id);
 
 /* Global interrupt enable */
 int32_t irq_global_enable(struct irq_ctrl_desc *desc);
 
 /* Global interrupt disable */
 int32_t irq_global_disable(struct irq_ctrl_desc *desc);
 
-/* Enable specific interrupt */
-int32_t irq_source_enable(struct irq_ctrl_desc *desc, uint32_t irq_id);
-
-/* Disable specific interrupt */
-int32_t irq_source_disable(struct irq_ctrl_desc *desc, uint32_t irq_id);
-
```

Comment:
Why did you decide to move those declarations?

Reply:
By mistake, will move them back